### PR TITLE
fix(mlx-lm): olmo 1b model

### PR DESCRIPTION
**Pull request by mzbac**
@awni While quantizing the Olmo 1B model, I just noticed that there are some bugs in the current implementation.

### CODE REVIEW

File: llms/mlx_lm/models/olmo.py [hunk 1]
```diff
@@ -24,6 +24,15 @@ class ModelArgs(BaseModelArgs):
     rope_theta: float = 10000
     rope_traditional: bool = False
     model_type: str = None
+    mlp_ratio: int = 4
+    weight_tying: bool = False
+
+    def __post_init__(self):
+        self.mlp_hidden_size = (
```
Comment (mzbac) [inline llms/mlx_lm/models/olmo.py:31]: https://github.com/allenai/OLMo/blob/main/olmo/model.py#L426-L428

File: llms/mlx_lm/models/olmo.py [hunk 2]
```diff
@@ -107,9 +117,12 @@ class Transformer(nn.Module):
     def __init__(self, args: ModelArgs):
         super().__init__()
         self.n_layers = args.n_layers
+        self.weight_tying = args.weight_tying
+
         self.wte = nn.Embedding(args.embedding_size, args.d_model)
         self.blocks = [TransformerBlock(args=args) for _ in range(args.n_layers)]
-        self.ff_out = nn.Linear(args.d_model, args.embedding_size, bias=False)
+        if not self.weight_tying:
```
Comment (mzbac) [inline llms/mlx_lm/models/olmo.py:124]: https://github.com/allenai/OLMo/blob/main/olmo/model.py#L1059-L1069

File: llms/mlx_lm/models/olmo.py [hunk 3]
```diff
@@ -130,7 +143,12 @@ def __call__(
         for e, block in enumerate(self.blocks):
             h, cache[e] = block(h, mask, cache[e])
 
-        return self.ff_out(self.norm(h)), cache
+        h = self.norm(h)
+
+        if self.weight_tying:
```
Comment (mzbac) [inline llms/mlx_lm/models/olmo.py:148]: https://github.com/allenai/OLMo/blob/main/olmo/model.py#L1300-L1305

**Review by awni — APPROVED** [review]
Thanks for fixing those!! My bad for missing them, I guess they are not used by the 7B model..

**Pull request merged**